lib/iris/cube.py

```diff
@@ -1639,7 +1639,7 @@ def dtype(self, dtype):
                 if not self.has_lazy_data():
                     emsg = 'Cube does not have lazy data, cannot set dtype.'
                     raise ValueError(emsg)
-                if dtype.kind != 'i':
+                if dtype.kind not in 'biu':
                     emsg = ('Can only cast lazy data to integral dtype, '
```

Comment:
This error message is now not technically true since you also check for boolean.